### PR TITLE
[INLONG-8369][Manager] Add MongoDB field type mapping strategy to improve usability

### LINKED ISSUE #8369: [Feature][Manager] Add MongoDB field type mapping strategy to improve usability
### Description

Inlong Mongo ExtractNode or LoadNode currently does not support some special field types, and it needs to be mapped to the corresponding inlong type to run correctly

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by chestnut-c**
[INLONG-8369][Manager] Add Mongo field type mapping strategy to improve usability

- Fixes #8369

### Motivation

Inlong Mongo ExtractNode or LoadNode currently does not support some special field types, add Mongo field type mapping strategy to mapping to the corresponding type to run correctly

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/DataNodeType.java [hunk 1]
```diff
@@ -35,4 +35,5 @@ public class DataNodeType {
     public static final String POSTGRESQL = "POSTGRESQL";
     public static final String ORACLE = "ORACLE";
     public static final String SQLSERVER = "SQLSERVER";
+    public static final String MONGO = "MONGO";
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/DataNodeType.java:38]: `MONGODB`

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/DataNodeType.java (hunk 1 above)
Comment (chestnut-c) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/DataNodeType.java]: > MONGODB

okk

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/MongoFieldTypeStrategy.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.apache.inlong.manager.common.fieldtype.FieldTypeMappingReader;
+
+import org.apache.commons.lang3.StringUtils;
+
+import static org.apache.inlong.manager.common.consts.InlongConstants.LEFT_BRACKET;
+
+/**
+ * The mongo field type mapping strategy
+ */
+public class MongoFieldTypeStrategy implements FieldTypeMappingStrategy {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/MongoFieldTypeStrategy.java:30]: `MongoDBFieldTypeStrategy`

File: inlong-manager/manager-pojo/src/test/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtilsTest.java [hunk 3]
```diff
@@ -84,4 +85,16 @@ public void testSQLServerFieldTypeInfo() {
         TypeInfo typeInfo = fieldInfo.getFormatInfo().getTypeInfo();
         Assertions.assertTrue(typeInfo instanceof StringTypeInfo);
     }
+
+    @Test
+    public void testMongoFieldTypeInfo() {
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/test/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtilsTest.java:90]: `testMongoDBFieldTypeInfo`

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/MongoProvider.java [hunk 4]
```diff
@@ -33,6 +35,8 @@
  */
 public class MongoProvider implements ExtractNodeProvider {
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/MongoProvider.java:36]: Can this class name change to `MongoDBProvider`?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/MongoFieldTypeStrategy.java (hunk 2 above)
Comment (chestnut-c) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/MongoFieldTypeStrategy.java]: > `MongoDBFieldTypeStrategy`

okk

File: inlong-manager/manager-pojo/src/test/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtilsTest.java (hunk 3 above)
Comment (chestnut-c) [reply inlong-manager/manager-pojo/src/test/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtilsTest.java]: > `testMongoDBFieldTypeInfo`

okk

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/MongoProvider.java (hunk 4 above)
Comment (chestnut-c) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/MongoProvider.java]: > Can this class name change to `MongoDBProvider`?

okk

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**